elkserver/scripts/alarm.py

```diff
@@ -21,21 +22,34 @@ def pprint(r):
  s = json.dumps(r, indent=2, sort_keys=True)
  return(s)
 
+def getValue(path, source):
+    p = path.split('.')
+    if p[0] in source:
+        if len(p) > 1:
+            return getValue('.'.join(p[1:]), source[p[0]])
+        else:
+            if p[0] == 'ip':
+                return source[p[0]][0]
+            else:
+                return source[p[0]]
+    else:
+        return None
+
```

Comment:
This function simplifies getting nested field values (returns None if the field doesn't exist)